connectors/sources/confluence.py

```diff
@@ -312,8 +341,230 @@ def get_default_configuration(cls):
+            )
+
+        return document
+
+    async def _user_access_control_doc(self, user):
+        """Generate a user access control document.
+
+        This method generates a user access control document based on the provided user information.
+        The document includes the user's account ID, prefixed account ID, prefixed account name,
+        a set of prefixed group IDs, and a set of prefixed role keys. The access control list is
+        then constructed using these values.
+
+        Args:
+            user (dict): A dictionary containing user information, such as account ID, display name, groups, and application roles.
+
+        Returns:
+            dict: A user access control document with the following structure:
+                {
+                    "_id": <account_id>,
+                    "identity": {
+                        "account_id": <_prefixed_account_id>,
+                        "display_name": <_preffixed_account_name>
+                    },
+                    "created_at": <iso_utc_timestamp>,
+                    ACCESS_CONTROL: [<_prefixed_account_id>, <_prefixed_group_ids>, <_prefixed_role_keys>]
+                }
+        """
+        account_id = user.get("accountId")
+        account_name = user.get("displayName")
+
+        _prefixed_account_id = _prefix_account_id(account_id=account_id)
+        _preffixed_account_name = _prefix_account_name(account_name=account_name)
+
+        _prefixed_group_ids = {
+            _prefix_group_id(group_id=group.get("groupId", ""))
+            for group in user.get("groups", {}).get("items", [])
+        }
+        _prefixed_role_keys = {
+            _prefix_role_key(role_key=role.get("key", ""))
+            for role in user.get("applicationRoles", {}).get("items", [])
+        }
+
+        user_document = {
+            "_id": account_id,
+            "identity": {
+                "account_id": _prefixed_account_id,
+                "display_name": _preffixed_account_name,
+            },
+            "created_at": iso_utc(),
         }
 
+        access_control = (
+            [_prefixed_account_id]
+            + list(_prefixed_group_ids)
+            + list(_prefixed_role_keys)
+        )
+
+        return user_document | self.access_control_query(access_control=access_control)
+
+    async def fetch_all_users(self):
+        async for users in self.confluence_client.api_call(
+            url=os.path.join(self.configuration["confluence_url"], URLS[USER])
+        ):
+            yield await users.json()
+
+    async def fetch_user(self, user_url):
+        async for user in self.confluence_client.api_call(
+            url=f"{user_url}&{USER_QUERY}"
+        ):
+            yield await user.json()
+
+    def _is_active_atlassian_user(self, user_info):
+        user_url = user_info.get("self")
+        if not user_url:
+            self._logger.debug(
+                f"Skipping {user_info.get('displayName', 'user')} as profile URL is not present."
```

Comment:
suggestion: I would introduce a variable for `user_info.get('displayName', 'user')` and reuse it throughout this method.